react/features/calendar-sync/functions.web.js

```diff
@@ -65,14 +70,25 @@ export function _fetchCalendarEntries(
  *
  * @param {string} calendarType - The calendar type api
  * as defined in CALENDAR_TYPE.
+ * @param {Object} store - The redux store.
  * @returns {Object}
  * @private
  */
-export function _getCalendarIntegration(calendarType) {
-    switch (calendarType) {
-    case CALENDAR_TYPE.GOOGLE:
-        return new GoogleCalendarApi(config.googleApiApplicationClientID);
-    case CALENDAR_TYPE.MICROSOFT:
-        return new MicrosoftCalendarApi(config.microsoftApiApplicationClientID);
+export function _getCalendarIntegration(calendarType, store) {
+    if (!calendarIntegrationInstance) {
+        switch (calendarType) {
+        case CALENDAR_TYPE.GOOGLE:
+            calendarIntegrationInstance
+                = new GoogleCalendarApi(
+                    config.googleApiApplicationClientID, store);
+            break;
+        case CALENDAR_TYPE.MICROSOFT:
+            calendarIntegrationInstance
+                = new MicrosoftCalendarApi(
+                    config.microsoftApiApplicationClientID, store);
+            break;
+        }
     }
+
+    return calendarIntegrationInstance;
```

Comment:
I'm unsure how I feel about the calendarIntegration Instance being stored in this file. I think it's fine but I also feel it might be something that could be stored in the redux state instead or held onto by some singleton.